src/com/xilinx/rapidwright/edif/EDIFNetlist.java

```diff
@@ -1611,14 +1624,17 @@ public void expandMacroUnisims(Series series) {
 		// Update all cell references to macro versions
 		for(EDIFLibrary lib : getLibraries()) {
 			boolean isHDILib = lib.isHDIPrimitivesLibrary(); 
-			for(EDIFCell cell : lib.getCells()) { 
+			for(EDIFCell cell : new ArrayList<>(lib.getCells())) { 
```

Comment:
I think I'm missing why this copy is necessary? It doesn't look like you add any cells to `lib` from what I can see in the diff context...